### ISSUE TITLE
Trouble loading checkpoint_best_ema.pt for RFDETRSegPreview

### ISSUE THREAD

**Issue by RM503** [roboflow/rf-detr#506]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I trained an RFDETRSegPreview model aim at identifying vehicles and materials at construction sites. I used the `checkpoint_best_ema.pt` on the same notebook for sample predictions and the segmented classes were good. Today, we I tried performing predictions using the checkpoint, the segmentation results look very incorrect. It even introduced labels that were not present in the training dataset. Since I am doing this on Google colab, when I try to load the checkpoint, it somehow defaults to using the pretrained weights. Could anyone tell me what the best way to proceed would be? Here is a minimal example

Here is a snippet from the predictions. For reference, there are 11 classes in my custom dataset. However, the model is predicting one with label 62, which should not happen.

```
...,
        [False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False]]]), confidence=array([0.31024548, 0.25558436, 0.18875965, 0.17413567, 0.15138197,
       0.14687121, 0.13785623, 0.13775767, 0.12168464, 0.12073859,
       0.11157419, 0.10148122, 0.10117555], dtype=float32), class_id=array([ 3,  7,  6,  8,  6,  8,  8, 62,  8,  8,  3,  3,  7]), tracker_id=None, data={}, metadata={})
```

### Environment

Environment:
- Google Colab
- L4 GPU
- RFDETR v1.3.0
- PyTorch v2.9.0+cu126

### Minimal Reproducible Example

```
import cv2
from rfdetr import RFDETRSegPreview
import supervision as sv
import torch 

IMG_PATH = "/content/drive/MyDrive/excavation_monitoring/sample_inputs/stock-photo-crawler-excavator-front-view-digging-on-demolition-site-2331610655.jpg"
OUTPUT_DIR = "/content/drive/MyDrive/excavation_monitoring/sample_inference"
CUSTOM_MODEL_WEIGHTS = "/content/drive/MyDrive/excavation_monitoring/training/output/checkpoint_best_ema.pth"

model = RFDETRSegPreview(
    pretrained_weights=CUSTOM_MODEL_WEIGHTS,
    device="cuda",
    resolution=432
)

img = cv2.imread(IMG_PATH)

with torch.no_grad():
    predictions = model.predict(img, threshold=0.1)

mask_annotator = sv.MaskAnnotator()
bbox_annotator = sv.BoxAnnotator()
annotated_frame = bbox_annotator.annotate(scene=img, detections=predictions)
annotated_frame = mask_annotator.annotate(scene=annotated_frame, detections=predictions)

predictions

cv2_imshow(annotated_frame)
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!
